### ISSUE TITLE
[Feature] Auto-sync of select folders

### ISSUE THREAD

**Issue by aneutron** [meganz/android#23]
Hey,

It would be very nice if we could have the possibility to choose some folders that would auto-sync on change or periodically.

Thanks.

**Comment by R34L1** [comment]
K

**Comment by blacktemplar** [comment]
Any updates on this? At least a one-way sync (from the cloud to the android phone) would be very nice to be able to use up to date files within applications.

**Comment by user8446** [comment]
Until this is a native feature, AutoSync for Mega does this https://play.google.com/store/apps/details?id=com.ttxapps.megasync&referrer=utm_source%3DMetaCtrlCom%26utm_medium%3Dreferral%26utm_campaign%3DMetaCtrlCom

**Comment by obedm503** [comment]
I would rather use the official app. I don't know if I can trust other clients. This is the one thing that the mobile app needs to be able to do. The desktop client already does this very well.

**Comment by kylejbrk** [comment]
Another vote for this feature to be added,

**Comment by PinkDev1** [comment]
> 
> 
> Until this is a native feature, AutoSync for Mega does this https://play.google.com/store/apps/details?id=com.ttxapps.megasync&referrer=utm_source%3DMetaCtrlCom%26utm_medium%3Dreferral%26utm_campaign%3DMetaCtrlCom

Yes, that third party app does syncronizations ok, but it has paid features, and it privacy policy is absolutely disgusting (https://metactrl.com/app-privacy/?app=megasync).

I'd rather wait for this feature to be implemented natively, and meanwhile support the meganz project.

**Comment by user8446** [comment]
Since your data is end-to-end encrypted,  there would be nothing to share outside of your type of device, etc.. No different than what MEGA can see themselves.
Maybe in the meantime a peer-to-peer sync would be good for you such as syncthing?

**Comment by PinkDev1** [comment]
@raquelgc6, @Yenel, @ash-mega, @ygaoy & @javiergm1983 
Any progress on this? This is the single most liked _issue_ in the proyect[1], and its been pending since 2017
We would really apreciate if you gave this a bit more importance 👀.

[1]: "is:issue is:open sort:reactions-+1-desc "
(https://github.com/meganz/android/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)

**Comment by javiergm1983** [comment]
We are working on add several new features like this.

**Comment by Pascalio** [comment]
Wonderful! Looking forward to this feature :+1:

**Comment by PJFonseca** [comment]
> We are working on add several new features like this.

So, more then 2 years have passed, but still nothing changed on this feature request.

People are using a third-party app to achieve this functionality.

I much prefer using the native app, for security reasons, but it seems Mega does give priority to this. I do have the premium plan, I do like the service, but this is a key feature for people.

Example: one must sync the keepass DB over phone and desktop. Mega is the secure way of storing that data BUT you need to use a third party app 🤷

Please, put some into it and get it done.

**Comment by MakeGnomeWebFast** [comment]
Guys, big pretty please get this done. Every user needs this feature.

**Comment by mohammadrafigh** [comment]
It's the only lack that makes me think if I should choose mega or not. I'm just investigating different solutions.

**Comment by mohammadrafigh** [comment]
Well I think it's not going to be implemented on iOS at all, and for Android it's not high priority due to this response from support team:

> Hello,
>
>
>
> MEGA mobile apps can't support sync of your entire file system other than Camera photos, videos and related subfolders such as "Twitter", "Screenshots", "Instagram", "WhatsApp" etc.
>
>
>
> ANDROID OS
>
> For Android 8 or greater, the sync feature on our Android app is in our to-do list. However, it is not a high priority project. It is not expected to be released in the short term.
>
>
>
> For now, you can use our desktop apps to synchronise other file types.
>
>
>
> For Android version prior to Android 8, unfortunately, this feature will not be available.
>
>
>
> The Android sync engine abuses the modification time of files to take sync decisions. There is a quite extended bug in many popular Android devices that makes it impossible to manage the modification time of files. Any synchronisation would modify the timestamps and files would get overwritten, re-uploaded over and over etc. Another problem would be the RAM usage and the disk usage. Many mobile users want MEGA to save disk space in their devices and the synchronization wouldn't allow that usage.
>
>
>
> More info here:
>
> https://code.google.com/p/android/issues/detail?id=34691
>
> https://code.google.com/p/android/issues/detail?id=18624
>
>
>
> Please upload other file types manually or use our desktop apps to synchronise other file types.
>
>
>
>
>
> iOS
>
> Due to Apple's operating system restrictions, we don't have access to any file system on your device but Photos.
>
> You can add individual files from other applications and select 'Open in' however that can't be done for folders.
>
> Please upload your file system data manually.
>
>
>
>
> Kind Regards,
> MEGA Support

**Comment by Yenel** [comment]
Hi @mohammadrafigh thank you for your feedback. We are already working on the sync feature, I cannot give you an ETA for this, but we are in the right way.

**Comment by mohammadrafigh** [comment]
Great. I got a pro plan to support this feature. Thank you

**Comment by Surendrajat** [comment]
Been paying some third party app for similar feature. It'll be nice to have this feature in official app after 5+ years.

**Comment by tazihad** [comment]
a two way sync for selected folders.
sync when change is made in either side.
files available offline.
external storage support.

Is all I am looking for.

**Comment by memtech3** [comment]
Howdy, what's the status of this feature? Does mega have a "bug bounty" style program where users can throw money behind features they'd like contributors to pick up?
